Document "Inso CLI on docker" (#66)
Kong/insomnia-docs: docs/inso-cli/inso-on-docker.md

---
layout: article-detail
title:  Inso CLI on Docker
category: "Inso CLI"
category-url: inso-cli
---

> > Note: this feature is still experimental and in active development.

Reviewer: ```suggestion
{:.alert .alert-primary}
> **Note**: this feature is still experimental and in active development.
```

> ## Building the `inso` container

Reviewer: ```suggestion
## Build the `inso` Container
```

Using active voice and avoiding "ing".

> 1. Clone the [insomnia repository](https://github.com/Kong/insomnia).

Reviewer: ```suggestion
1. Clone the [Insomnia repository](https://github.com/Kong/insomnia).
```

2. Install [Docker](https://docs.docker.com/get-docker/).
3. Run `docker build -t insomnia-inso:dev -f inso-dev.Dockerfile .`.

> ## Using the `inso` container

Reviewer: ```suggestion
## Use the `inso` Container
```

> In order to run Insomnia specs in a container, you'll need to mount the specs folder on your host machine to a given folder on the container. Examples:

Reviewer: ```suggestion
To run Insomnia specs in a Docker container, mount the specs folder on your host machine to a given folder on the container. Examples:
```

Increase casual language and specificity.

> - Mounting an Insomnia git sync repository folder to a folder on the container:

Reviewer: ```suggestion
- Mount an Insomnia git sync repository folder to a folder on the container:
```

```shell
cd <your-git-sync-repo-folder>

docker run -it --rm -v $(pwd):/var/temp insomnia-inso:dev run test -w /var/temp
```

> - Mounting the Insomnia Application Data folder:

Reviewer: ```suggestion
- Mount the Insomnia Application Data folder:
```

```shell
# On macOS:

> 

Reviewer: ```suggestion
```

docker run -v $HOME/Library/Application\ Support/Insomnia:/var/temp -it --rm insomnia-inso:dev run test --src /var/temp

# On Linux:
docker run -v $HOME/.config/Insomnia:/var/temp -it --rm insomnia-inso:dev run test --src /var/temp

# On Windows (using Docker for Windows and WSL):
docker run -v /mnt/c/Users/<your_username>/AppData/Roaming/Insomnia:/var/temp -it --rm insomnia-inso:dev run test --src /var/temp
```

> - Mounting the folder where you keep an Insomnia v4 export:

Reviewer: ```suggestion
- Mount the folder where you keep an Insomnia v4 export:
```
